relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -1,93 +1,141 @@
+                    condition: None,
+                },
+                TagSpec {
+                    key: "span.status".into(),
+                    field: Some("span.data.span.status".into()),
+                    value: None,
+                    condition: None,
+                },
+                TagSpec {
+                    key: "span.status_code".into(),
+                    field: Some("span.data.span.status_code".into()),
+                    value: None,
+                    condition: None,
+                },
+                TagSpec {
+                    key: "span.system".into(),
+                    field: Some("span.data.span.system".into()),
+                    value: None,
+                    condition: None,
+                },
+                TagSpec {
+                    key: "transaction.method".into(),
+                    field: Some("span.data.transaction.method".into()),
+                    value: None,
+                    condition: None,
+                },
+                TagSpec {
+                    key: "transaction.op".into(),
+                    field: Some("span.data.transaction.op".into()),
+                    value: None,
+                    condition: None,
+                },
+            ],
+        }],
+    });
 
-/// Extracts metrics from the spans of the given transaction, and sets common
-/// tags for all the metrics and spans. If a span already contains a tag
-/// extracted for a metric, the tag value is overwritten.
-pub(crate) fn extract_span_metrics(
-    aggregator_config: &AggregatorConfig,
-    event: &Event,
-) -> Result<Vec<Metric>, ExtractMetricsError> {
+/// Extracts metrics from the spans of the given transaction.
+pub(crate) fn extract_span_metrics(event: &Event) -> Result<Vec<Metric>, ExtractMetricsError> {
     // TODO(iker): measure the performance of this whole method
     let mut metrics = Vec::new();
-
-    if event.ty.value() != Some(&EventType::Transaction) {
-        return Ok(metrics);
-    }
-    let (Some(&_start), Some(&end)) = (event.start_timestamp.value(), event.timestamp.value()) else {
-        relay_log::debug!("failed to extract the start and the end timestamps from the event");
-        return Err(ExtractMetricsError::MissingTimestamp);
-    };
-
-    let Some(timestamp) = UnixTimestamp::from_datetime(end.into_inner()) else {
-        relay_log::debug!("event timestamp is not a valid unix timestamp");
-        return Err(ExtractMetricsError::InvalidTimestamp);
-    };
-
-    // Validate the transaction event against the metrics timestamp limits. If the metric is too
-    // old or too new, we cannot extract the metric and also need to drop the transaction event
-    // for consistency between metrics and events.
-    if !aggregator_config.timestamp_range().contains(&timestamp) {
-        relay_log::debug!("event timestamp is out of the valid range for metrics");
-        return Err(ExtractMetricsError::InvalidTimestamp);
-    }
```

Comment:
@iker-barriocanal is it ok if I remove this validation? It is already done for transactions in a different place (see below)

https://github.com/getsentry/relay/blob/97a9a29e7503385e4224ece3c0e09938b87e39bb/relay-server/src/metrics_extraction/transactions/mod.rs#L263-L273

Reply:
Yep 👍